packages/report-server/src/reportBuilder/transform/functions/insertColumns.ts

```diff
@@ -36,9 +38,7 @@ const insertColumns = (table: TransformTable, params: InsertColumnsParams, conte
       const columnName = parser.evaluate(key);
       const columnValue = parser.evaluate(expression);
       if (!newColumns[columnName]) {
-        newColumns[columnName] = table.hasColumn(columnName)
-          ? table.getColumnValues(columnName) // Upserting a column, so fill with current column values
-          : new Array(table.length()).fill(undefined); // Creating a new column, so fill with undefined
+        newColumns[columnName] = new Array(table.length()).fill(undefined);
```

Comment:
Since we now re-insert the skipped rows, there's no need for the conditional upsert logic, just overwrite everything and let the skipped rows re-insert themselves if they exist.